### PR TITLE
MBS-10865 slack api update; use channel id's instead of names in config

### PR DESCRIPTION
**Pull request by dsvoronin**
This is a API breakage: SlackChannel is used in public configs

Users should specify channel id's instead of names,
for example: `C01D88JT6CX` instead of `#android-speed-test-integration`
implementation now more straightforward, no global searches

Why not just use `conversations.list`? 
It's [hard limited](https://api.slack.com/methods/conversations.list#arg_limit) with up to 1000 channels, which is not enough in our case; previous method was not limited

### CODE REVIEW

File: subprojects/gradle/lint-report/build.gradle.kts [hunk 1]
```diff
@@ -41,7 +41,7 @@ gradlePlugin {
 }
 
 tasks.named<Test>("integrationTest").configure {
-    applyOptionalSystemProperty("avito.slack.test.channel")
+    applyOptionalSystemProperty("avito.slack.test.channelid")
```
Comment (eugene-krivobokov) [inline subprojects/gradle/lint-report/build.gradle.kts]:
```suggestion
    applyOptionalSystemProperty("avito.slack.test.channelId")
```
Can we use camelCase? It's easier for reading

File: subprojects/gradle/slack-test-fixtures/src/main/kotlin/com/avito/slack/model/StubFoundMessage.kt [hunk 2]
```diff
@@ -6,12 +6,12 @@ fun FoundMessage.Companion.createStubInstance(
     timestamp: String = "",
     emoji: String? = null,
     author: String? = null,
-    channel: SlackChannel = SlackChannel("#")
+    channelId: SlackChannelId = SlackChannelId("C01D88JT6CX")
```
Comment (eugene-krivobokov) [inline subprojects/gradle/slack-test-fixtures/src/main/kotlin/com/avito/slack/model/StubFoundMessage.kt]: Duplicate defaults from subprojects/gradle/lint-report/src/test/kotlin/com/avito/android/lint/ValidGradleTaskNameTest.kt
Let's reuse the value

File: subprojects/gradle/slack/src/main/kotlin/com/avito/slack/SlackClient.kt [hunk 3]
```diff
@@ -5,29 +5,29 @@ package com.avito.slack
...
-        channel: SlackChannel,
+        channelId: SlackChannelId,
         text: String,
         messageTimestamp: String
     ): Result<SlackMessage>
 
-    fun findMessage(channel: SlackChannel, predicate: SlackMessageUpdateCondition): Result<FoundMessage>
+    fun findMessage(
+        channelId: SlackChannelId,
+        predicate: SlackMessageUpdateCondition
```
Comment (eugene-krivobokov) [inline subprojects/gradle/slack/src/main/kotlin/com/avito/slack/SlackClient.kt]: I didn't get API
The method is named `findMessage` but updates a message.
I'd like to make API more obvious:
- Name it update...
- Or move responsibility for an update to client

File: subprojects/gradle/slack/src/main/kotlin/com/avito/slack/SlackClient.kt [hunk 4]
```diff
@@ -5,29 +5,29 @@ package com.avito.slack
...
 import com.github.seratch.jslack.api.methods.request.chat.ChatUpdateRequest
-import com.github.seratch.jslack.api.model.Channel
+import com.github.seratch.jslack.api.methods.request.conversations.ConversationsHistoryRequest
 import java.io.File
 
 interface SlackClient : SlackMessageSender, SlackFileUploader {
 
     fun updateMessage(
-        channel: SlackChannel,
+        channelId: SlackChannelId,
```
Comment (sergioboi) [inline subprojects/gradle/slack/src/main/kotlin/com/avito/slack/SlackClient.kt]: The idea - get Channel info by id and compare actual and expected Channel name

File: subprojects/gradle/lint-report/build.gradle.kts (hunk 1 above)
Comment (dsvoronin) [reply subprojects/gradle/lint-report/build.gradle.kts]: yes, will fix

File: subprojects/gradle/lint-report/src/main/kotlin/com/avito/android/lint/slack/LintReportToSlackTaskFactory.kt [hunk 5]
```diff
@@ -36,9 +37,9 @@ class LintReportToSlackTaskFactory(
     /**
      * To be used in CiStep, because slack channel only known from there
      */
-    fun registerLintReportToSlackTask(slackChannel: String): TaskProvider<LintSlackReportTask> {
+    fun registerLintReportToSlackTask(channel: SlackChannel): TaskProvider<LintSlackReportTask> {
 
-        val taskName = "lintReportTo${slackChannel.validInGradleTaskName()}"
+        val taskName = "lintReportTo${channel.id.validInGradleTaskName()}"
```
Comment (eugene-krivobokov) [inline subprojects/gradle/lint-report/src/main/kotlin/com/avito/android/lint/slack/LintReportToSlackTaskFactory.kt]:
```suggestion
        val taskName = "lintReportTo${channel.name.validInGradleTaskName()}"
```
?

File: subprojects/gradle/test-summary/src/main/kotlin/com/avito/test/summary/TestSummarySender.kt [hunk 6]
```diff
@@ -89,7 +89,7 @@ internal class TestSummarySenderImpl(
                     ).onSuccess { message ->
                         sendMessage(
                             SlackSendMessageRequest(
-                                channel = channel,
+                                id = channel,
```
Comment (eugene-krivobokov) [inline subprojects/gradle/test-summary/src/main/kotlin/com/avito/test/summary/TestSummarySender.kt:92]: Can keep `channel` name for variables and properties

File: subprojects/gradle/slack/src/main/kotlin/com/avito/slack/model/SlackSendMessageRequest.kt [hunk 7]
```diff
@@ -1,7 +1,7 @@
 package com.avito.slack.model
 
 data class SlackSendMessageRequest(
-    val channel: SlackChannel,
+    val id: SlackChannel,
```
Comment (eugene-krivobokov) [inline subprojects/gradle/slack/src/main/kotlin/com/avito/slack/model/SlackSendMessageRequest.kt]: Can't we keep the `channel`?

File: subprojects/gradle/slack/src/main/kotlin/com/avito/slack/SlackClient.kt (hunk 3 above)
Comment (dsvoronin) [reply subprojects/gradle/slack/src/main/kotlin/com/avito/slack/SlackClient.kt]: renamed SlackMessageUpdateCondition to Predicate, now it all makes sense

File: subprojects/gradle/slack/src/main/kotlin/com/avito/slack/SlackClient.kt (hunk 4 above)
Comment (dsvoronin) [reply subprojects/gradle/slack/src/main/kotlin/com/avito/slack/SlackClient.kt]: ID should be stable for public channels, names could vary
I don't know if it make sense to compare names and id
i only see value of providing name as a descriptive field (readability of configs);
 but if channel name was changed this description will be outdated; so comment should be enough;

i'll add name for now; let's decide if it was worth - later

name and id could be compared via [conversations.info](https://api.slack.com/methods/conversations.info)

File: subprojects/gradle/lint-report/src/main/kotlin/com/avito/android/lint/slack/LintReportToSlackTaskFactory.kt (hunk 5 above)
Comment (dsvoronin) [reply subprojects/gradle/lint-report/src/main/kotlin/com/avito/android/lint/slack/LintReportToSlackTaskFactory.kt]: thanks!

**Review by eugene-krivobokov — APPROVED** [review]

**Pull request merged**